From 0a8cef484174aae5c1b7be6710f31a643e7d7197 Mon Sep 17 00:00:00 2001
From: Yuri <yuri@tsoft.com>
Date: Sun, 26 Nov 2017 13:11:18 -0800
Subject: [PATCH] cmake: fix ntk include path, add cairo library

Moved ntk include path earlier to avoid conflict with fltk
Added missing cairo to CMakeLists.txt

Fixes #19.
---
 CMakeLists.txt | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f9f0702..7c0d989 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,13 +15,17 @@ find_package(PkgConfig)
 
 #SET(CMAKE_INSTALL_PREFIX "/usr" )
 
-pkg_check_modules(LV2 lv2 REQUIRED)
-include_directories( ${LV2_INCLUDE_DIRS}  )
-
 pkg_check_modules(NTK ntk REQUIRED)
 include_directories( ${NTK_INCLUDE_DIRS}  )
 link_directories   ( ${NTK_LIBRARY_DIRS}  )
 
+pkg_check_modules(LV2 lv2 REQUIRED)
+include_directories( ${LV2_INCLUDE_DIRS}  )
+
+pkg_check_modules(CAIRO cairo REQUIRED)
+include_directories( ${CAIRO_INCLUDE_DIRS}  )
+link_directories   ( ${CAIRO_LIBRARY_DIRS}  )
+
 SET(CMAKE_SHARED_LINKER_FLAGS "-fPIC -shared -Wl,-z,nodelete -Wl,--no-undefined")
 
 IF(RELEASE_BUILD)
@@ -81,6 +85,8 @@ ADD_LIBRARY(sorcer SHARED ${sources})
 target_link_libraries( sorcer ${NTK_LIBRARIES}     )
 #target_link_libraries( sorcer ${SNDFILE_LIBRARIES} )
 
+target_link_libraries( sorcer ${CAIRO_LIBRARIES}     )
+
 # Remove "lib" part before name (sorcer.so, not libsorcer.so)
 set_target_properties(sorcer PROPERTIES PREFIX "")
 
